geomstats/learning/polynomial_regression.py

```diff
@@ -0,0 +1,572 @@
+    - :math:`X` is the input, :math:`Y` is the target.
+
+    The polynomial regression method:
+
+    - estimates :math:`\beta_0`, and each :math:`\beta_k`,
+    - predicts :math:`\hat{y}` from input :math:`X`.
+
+    Parameters
+    ----------
+    space : Manifold
+        Manifold.
+    metric : RiemannianMetric
+        Riemannian metric.
+    center_X : bool
+        Subtract mean to X as a preprocessing.
+    order : int
+        Order of polynomial to fit. Mandatory, maximum is 5 (for now)
+    method : str, {\'extrinsic\', \'riemannian\'}
+        Gradient descent method.
+        Optional, default: extrinsic.
+    max_iter : int
+        Maximum number of iterations for gradient descent.
+        Optional, default: 100.
+    init_step_size : float
+        Initial learning rate for gradient descent.
+        Optional, default: 0.1
+    tol : float
+        Tolerance for loss minimization.
+        Optional, default: 1e-5
+    verbose : bool
+        Verbose option.
+        Optional, default: False.
+    initialization : str or array-like,
+        {'random', 'data', 'frechet', warm_start'}
+        Initial values of the parameters for the optimization,
+        or initialization method.
+        Optional, default: 'random'
+    regularization : float
+        Weight on the constraint for the intercept to lie on the manifold in
+        the extrinsic optimization scheme. An L^2 constraint is applied.
+        Optional, default: 1.
+    """
+
+    def __init__(
+        self,
+        space,
+        order,
+        metric=None,
+        center_X=True,
+        method="extrinsic",
+        max_iter=100,
+        init_step_size=0.1,
+        tol=1e-5,
+        verbose=False,
+        initialization="random",
+        regularization=1.0,
+    ):
+        if metric is None:
+            metric = space.metric
+        self.metric = metric
+        self.space = space
+
+        # For now, only allow for fitting through quintic polynomials
+        geomstats.errors.check_parameter_accepted_values(
+            order, "order", [i + 1 for i in range(0, 5)]
+        )
+        self.order = order
+
+        self.intercept_ = None
+        self.coef_ = None
+        self.center_X = center_X
+        self.mean_ = None
+        self.training_score_ = None
+        geomstats.errors.check_parameter_accepted_values(
+            method, "method", ["extrinsic", "riemannian"]
+        )
+        self.method = method
+        self.max_iter = max_iter
+        self.verbose = verbose
+        self.init_step_size = init_step_size
+        self.tol = tol
+        self.initialization = initialization
+        self.regularization = regularization
+
+    def _model(self, X, coef, intercept):
+        """Compute the generative model of the polynomial regression.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape=[...,}]
+            Training input samples.
+        coef : array-like, shape=[..., {dim, [n,n]}]
```

Comment:
On line 156 below, it says "Coef is of shape k x dim (where dim can be multidimensional)" which contradicts the shape of `coef` given in the docstring --> Unify.
`shape=[..., order, {dim, [n, n]}]`

Additionally, in the docstirng, use "order" to refer to the order of the polynomial, not "K" nor "k".